src/main/java/com/github/jasminb/jsonapi/ResourceConverter.java

```diff
@@ -351,12 +360,14 @@ private void handleRelationships(JsonNode source, Object object, Map<String, Obj
 
 					// Use resolver if possible
 					if (resolveRelationship && resolver != null && relationship.has(LINKS)) {
-						JsonNode self = relationship.get(LINKS).get(SELF);
+						String relType = RELATIONSHIP_RELTYPE_MAP.get(object.getClass())
```

Comment:
You can use `relationshipField` to get the `Relationship` from the `FIELD_RELATIONSHIP_MAP`:

```
String relType = FIELD_RELATIONSHIP_MAP.get(relationshipField).getRelType();
```